### DISCUSSION TITLE
kill() command doesn't times

### DISCUSSION THREAD

**Discussion by DarkiCreepKato** [HaxeFlixel/flixel#2837] [Q&A] [answered]
I was trying make a image appear and disappear with time with FromTime(t:Float):date, it first worked with add sprite (it timed the aggregation of the sprite)
But when i tried timing the kill() and other comands like the revive() and visible:Bool and alive:Bool it didn't time and just make the sprite disappear since the start, maybe its a bug or something but im too lazy to go get a different haxe version or wait till they update haxeflixel, maybe I'll look for how to fix it first

**Comment by Geokureli** [comment]
I'm not sure what you mean by "didn't time" or "doesn't times" I'm also not familiar with `fromTime(t:Float):date`. Not sure if english is a second language for you. It would be very helpful if you shared a snippet of code showing exactly what you're trying to do

**Reply by DarkiCreepKato** [reply]
a snippet of my code or someone else's?  And yeah i speak Spanish, but english is a second language for me what i meant is that it didn't did not wait for time its that i tried making the code do kill, it didn't waited the time i wrote and it kill immediately when i run the exe :(

**Reply by Geokureli** [reply]
I can't help unless I see your code

**Reply by DarkiCreepKato** [reply]
ok here it is, this is playstate:

package;

import flixel.FlxSprite;
import flixel.FlxState;
import flixel.util.FlxTimer;

class PlayState extends FlxState
{
	var PinkScreen:FlxSprite;

	override public function create()
	{
		super.create();
		PinkScreen = new FlxSprite();
		PinkScreen.loadGraphic(AssetPaths.PinkStartGame__png);

		PinkScreen.x = 0;
		PinkScreen.y = 0;
		Date.fromTime(2000);
		add(PinkScreen);
		PinkScreen.visible = true;
		Date.fromTime(2000);
		PinkScreen.visible = false;
	}

	override public function update(elapsed:Float)
	{
		super.update(elapsed);
	}
}

**Reply by Geokureli** [reply, marked as answer]
`Date.fromTime` does not do what you think. You want to use [FlxTimers](https://api.haxeflixel.com/flixel/util/FlxTimer.html)

example:
```hx
override public function create()
{
	super.create();
	PinkScreen = new FlxSprite();
	PinkScreen.loadGraphic(AssetPaths.PinkStartGame__png);

	PinkScreen.x = 0;
	PinkScreen.y = 0;
	// wait 2 seconds then add PinkScreen and show it
	new FlxTimer().start(2.0, 
		function(t:FlxTimer)
		{
			add(PinkScreen);
			PinkScreen.visible = true; // note: this line does nothing because visible is true, by default
		}
	);
	// wait 4 seconds then hide it (this will happen 2 seconds after it is first shown)
	new FlxTimer().start(4.0, 
		function(t:FlxTimer)
		{
			PinkScreen.visible = false;
		}
	);
}
```

**Reply by DarkiCreepKato** [reply]
oh okay thank you
